jobs/batch_jobs/opportunity_jobs.py

```diff
@@ -0,0 +1,188 @@
+                    'models': exploration_opportunity_summary_model_list
+                }
+        except Exception as e:
+            return {
+                    'status': 'FAILURE',
+                    'job_result': job_run_result.JobRunResult(stderr='FAILURE: %s' % e),
+                    'models': []
+                }
+        
+
+    def run(self) -> beam.PCollection:
+        """Returns a PCollection of 'SUCCESS' or 'FAILURE' results from
+        generating ExplorationOpportunitySummaryModel.
+
+        Returns:
+            PCollection. A PCollection of 'SUCCESS' or 'FAILURE' results from
+            generating ExplorationOpportunitySummaryModel.
+        """
+        # pre start hook
+        exp_oppurtunity_models = (
+            opportunity_models.ExplorationOpportunitySummaryModel.get_all(
+                        include_deleted=False))
+        (
+            self.pipeline 
+            | beam.Create([model.key for model in exp_oppurtunity_models])
+            | 'Delete all models' >> ndb_io.DeleteModels()
+        )
+
+        topics = (
+            self.pipeline
+            | 'Get all non-deleted topic models' >> (
+                ndb_io.GetModels( # type: ignore[no-untyped-call]
+                    topic_models.TopicModel.get_all(include_deleted=False)))
+            | 'Get topic from model' >> beam.Map(
+                topic_fetchers.get_topic_from_model)
+        )
+
+        story_ids_to_story = (
+            self.pipeline
+            | 'Get all non-deleted story models' >> (
+                ndb_io.GetModels( # type: ignore[no-untyped-call]
+                    story_models.StoryModel.get_all(include_deleted=False)))
+            | 'Get story from model' >> beam.Map(
+                story_fetchers.get_story_from_model)
+            | 'Combine stories and ids' >> beam.Map(lambda story: (story.id, story))
+        )
+
+        exp_ids_to_exp = (
+            self.pipeline
+            | 'Get all non-deleted exp models' >> (
+                ndb_io.GetModels( # type: ignore[no-untyped-call]
+                    exp_models.ExplorationModel.get_all(include_deleted=False)))
+            | 'Get exploration from model' >> beam.Map(
+                exp_fetchers.get_exploration_from_model)
+            | 'Combine exploration and ids' >> beam.Map(lambda exp: (exp.id, exp))
+        )
+
+
+        stories_dict = beam.pvalue.AsDict(story_ids_to_story)
+        exploration_dict = beam.pvalue.AsDict(exp_ids_to_exp)
+
+        output = (
+            topics
+            | beam.Map(
+                self._regenerate_opportunities_related_to_topic, stories_dict=stories_dict, exploration_dict=exploration_dict)
+        )
+
+        (
+            output 
+            | 'Filter the results with SUCCESS status' >> beam.Filter(
+                lambda result: result['status'] == 'SUCCESS')
+            | 'Fetch the models to be put' >> beam.FlatMap(lambda result: result['models'])
+            | ndb_io.PutModels()
+        )
```

Comment:
@vojtechjelinek I think the models are not being stored because self.pipeline is not being used. Am I correct?
If so how should I combine both any suggestions?
(The models are not being stored in the database and I have verified it manually and using testing)